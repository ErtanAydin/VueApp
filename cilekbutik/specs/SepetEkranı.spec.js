import { mount, createLocalVue, RouterLinkStub } from "@vue/test-utils";
import SepetEkranı from "@/components/SepetEkranı";
import Vuex from "vuex"

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  state: {
    Sepet: []
  },
  actions: {
    Sepetfetch(){
      let Sepet = [
        {
            id:0,
            url:'https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg',
            name:'Lakers Sweat',
            fiyat:55.00,
            barkod:"6242005630101",
            kategori:"Ceket",
            beden:'M',
            renk:'TURUNCU',
            tutar:55.00,
            adet:10,
            counter:1,
        }]
        this.state.Sepet=Sepet
    }
  }
})

describe("methods", () => {
  test("AdetArttır", () => {
    const { AdetArttır } = SepetEkranı.methods
    let Urun = { counter : 0 }
    AdetArttır(Urun)
    expect(Urun.counter).toBe(1);
  });

  test("UrunSil", () => {
    const wrapper = mount(SepetEkranı, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })
    wrapper.vm.UrunSil()
    expect(store.state.sepet).toBe(undefined)
  })

  test("AdetAzalt", () => {
    const { AdetAzalt } = SepetEkranı.methods
    let Urun = { counter : 2 }
    AdetAzalt(Urun)
    expect(Urun.counter).toBe(1);
    AdetAzalt(Urun)
    expect(Urun.counter).toBe(1);
  });
});

describe("SepetEkranı", () => {
  test("mounts properly", () => {
    const wrapper = SepetEkranı;
    expect(wrapper).toBeTruthy();
  });

  test("gerçek Vuex Store kullanarak SepetEkranı render edildi", () => {
    const wrapper = mount(SepetEkranı, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })

    expect(wrapper.html()).toMatchSnapshot()
  });
});

describe('created', () => {
  test('created test', () => {
    expect(store.state.Sepet).not.toBe(undefined)
  })
})

describe('computed', () => {
  test('computed test', () => {
    expect(store.state.Sepet).not.toBe(undefined)
  })
  test('GenelToplamHesaplama test', () => {
    const wrapper = mount(SepetEkranı, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })
    expect(wrapper.vm.GenelToplamHesaplama).toBe(55)
  })
  test('GenelToplamHesaplamaKDV test', () => {
    const wrapper = mount(SepetEkranı, { 
      store, 
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub
      } 
    })
    expect(wrapper.vm.GenelToplamHesaplamaKDV).toBe(55)
  })
})