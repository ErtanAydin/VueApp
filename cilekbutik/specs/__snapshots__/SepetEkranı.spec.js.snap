// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SepetEkranı gerçek Vuex Store kullanarak SepetEkranı render edildi 1`] = `
<section class="SepetEkranıSection">
  <div class="SepetEkranı">
    <p class="SepetListesiYazı">Sepet Listesi</p>
    <div class="SepetEkranıBirim">
      <p class="SepetEkranıUrun">Ürün</p>
      <p class="SepetEkranıFiyat">Fiyat</p>
      <p class="SepetEkranıAdet">Adet</p>
      <p class="SepetEkranıTutar">Tutar</p>
    </div>
    <div class="SepetUrunler">
      <div sepetfiyatkonum=""><b class="UrunAdetFiyat"> 55,00 TL </b> <button class="AdetAzaltButonu">-</button> <input class="AdetInput"> <button class="AdetArttırButonu">+</button> <b class="Tutar"> 55,00 TL </b></div> <a><img src="https://www.cilekbutik.net/cdn/1/300/500/images/urunler/5fa57c217b440-1353.jpg" alt="" class="UrunImage"></a>
      <div class="konum"><a class="LINK"><b alt="" class="UrunIsim">
            Lakers Sweat
          </b></a> <b class="UrunBeden">
          BEDEN:
          <span> M </span></b> <b class="UrunRenk">
          Renk:
          <span> TURUNCU </span></b> <button class="UrunSilButton">Sil</button></div>
    </div>
    <div class="AlışverişeDevamEtDiv"><a><button class="AlışverişeDevamEtButon">
          &lt; Alışverişe Devam Et
        </button></a></div>
  </div>
  <div class="ŞiparişÖzeti">
    <h3 class="ŞiparişÖzetiYazı">Sipariş Özeti</h3> <b class="ŞiparişÖzetiAra"> &nbsp;Ürünler Toplamı <b class="ToplamFiyat"> 55,00 TL</b></b>
    <div class="ŞiparişÖzetiGenel"><b> &nbsp;
        Genel Toplam
        <b class="GenelToplamFiyat"> 55,00 TL</b> <br> &nbsp;(KDV Dahil)
      </b></div> <a><button class="AlışverişiTamamlaButonu"> Alışverişi Tamamla &gt; </button></a>
  </div>
</section>
`;
